src/proof/common.ts

```diff
@@ -227,14 +231,72 @@ export const transformQueryValueToBigInts = async (
   value: unknown,
   ldType: string
 ): Promise<bigint[]> => {
-  const values: bigint[] = new Array<bigint>(64).fill(BigInt(0));
+  const values: bigint[] = [];
 
   if (Array.isArray(value)) {
     for (let index = 0; index < value.length; index++) {
-      values[index] = await Merklizer.hashValue(ldType, value[index]);
+      values.push(await Merklizer.hashValue(ldType, value[index]));
     }
   } else {
     values[0] = await Merklizer.hashValue(ldType, value);
   }
   return values;
 };
+
+const queryTransformV3Circuit = (query: Query): Query => {
+  if (query.operator === Operators.SD) {
+    return new Query({
+      ...query,
+      values: new Array(defaultValueArraySize).fill(0)
```

Comment:
are we sure that we rewrite values? If I'm not mistaken the value itself has already been written to the query.values